source/uwp/Renderer/lib/AdaptiveImageRenderer.cpp

```diff
@@ -533,12 +492,133 @@ namespace AdaptiveCards::Rendering::Uwp
+                    auto strongImageContainer = imageContainerRef.get();
+
+                    if (strongThis && strongImageSource && strongStream && strongUiElement && strongParentElement && strongImageContainer)
+                    {
+                        if (status == winrt::AsyncStatus::Completed)
+                        {
+                            auto bytes = result.GetResults();
+                            auto svgText = streamDataReader.ReadString(bytes);
+
+                            auto svgDocument = winrt::XmlDocument();
+                            svgDocument.LoadXml(svgText);
+
+                            strongThis->ParseXmlForHeightAndWidth(svgDocument, strongImageSource);
+                        }
+
+                        // Now that we've parsed the height, we can set the image source
+                        strongThis->SetSvgImageSourceAsync(
+                            strongImageSource, strongStream, strongUiElement, isAutoSize, strongParentElement, strongImageContainer, isVisible);
+                    }
+                });
+        }
+        else
+        {
+            auto setSourceAction = imageSource.as<winrt::BitmapImage>().SetSourceAsync(stream);
+
+            setSourceAction.Completed([weakThis = this->get_weak(), uiElement, isAutoSize, parentElement, imageContainer, isVisible](
+                auto const& /*operation*/, auto status)
+                {
+                    if (status == winrt::AsyncStatus::Completed && isAutoSize)
+                    {
+                        if (auto strongThis = weakThis.get())
+                        {
+                            strongThis->SetAutoSize(uiElement, parentElement, imageContainer, isVisible, false /* imageFiresOpenEvent */);
+                        }
+                    }
+                });
+        }
+    }
+
+    winrt::IAsyncAction render_xaml::XamlBuilder::SetSvgUriSource(winrt::SvgImageSource const imageSource,
+                                                                  winrt::Uri const uri)
+    {
+        co_await winrt::resume_foreground(imageSource.Dispatcher());
+        imageSource.UriSource(uri);
+    }
+
+    template<typename T, typename S>
+    winrt::IAsyncAction render_xaml::XamlBuilder::SetSvgImageSourceAsync(winrt::SvgImageSource const imageSource,
+                                                                         S const stream,
+                                                                         T const uiElement,
+                                                                         bool isAutoSize,
+                                                                         winrt::IInspectable const parentElement,
+                                                                         winrt::IInspectable const imageContainer,
+                                                                         bool isVisible)
+    {
+        co_await winrt::resume_foreground(imageSource.Dispatcher());
```

Comment:
>co_await [](http://example.com/codeflow?start=8&length=8)

Once you co_await there is nothing keeping `this` alive.  You should either capture a weakRef before the await to pass into the lambda, or call `get_strong` and save it in a variable so this is not destroyed while the await is happening.